src/trie/trie.c

```diff
@@ -3,23 +3,32 @@
 #include "util/bsearch.h"
 #include "sparse_vector.h"
 #include "redisearch.h"
+#include "rmutil/rm_assert.h"
 #include "util/arr.h"
+#include "config.h"
 
 size_t __trieNode_Sizeof(t_len numChildren, t_len slen) {
   return sizeof(TrieNode) + numChildren * sizeof(TrieNode *) + sizeof(rune) * (slen + 1);
 }
 
-TriePayload *triePayload_New(const char *payload, uint32_t plen);
-
 // Allocate a new trie payload struct
-inline TriePayload *triePayload_New(const char *payload, uint32_t plen) {
+static inline TriePayload *triePayload_New(const char *payload, uint32_t plen) {
 
   TriePayload *p = rm_malloc(sizeof(TriePayload) + sizeof(char) * (plen + 1));
+  p->data[plen] = '\0';
```

Comment:
This is a bug no?

Reply:
This is to silence a warning that came up in the cpp tests added.
It NULLs the `+ 1` byte from line 17.

Reply:
What warning? If we do not really need it to be NULL terminated then why adding it?

Reply:
Commented out line but still don't see warning.

Reply:
The warning is in the sanitizer with the cpp tests running ASSERT_STREQ.